### PR TITLE
kubevirt CI scripts

### PR DESCRIPTION
**Pull request by davidvossel**
This introduces all the changes made in the kubevirt/cluster-api-provider-kubevirt fork that were developed to help wire capk to kubevirtci. 

```release-note
NONE
```

### CODE REVIEW

File: config/default/manager_image_patch.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: gcr.io/k8s-staging-cluster-api/capk-manager:dev
+      - image: registry:5000/capk-manager-amd64:dev
```
Comment (davidvossel) [inline config/default/manager_image_patch.yaml:11]: This change is likely problematic. I think we'll need a different kustomize layer for kubevirtci

File: templates/cluster-template.yaml [hunk 2]
```diff
@@ -68,10 +68,10 @@ spec:
       imageRepository: ${IMAGE_REPO}
     initConfiguration:
       nodeRegistration:
-        criSocket: /var/run/containerd/containerd.sock
+        criSocket: "${CRI_PATH}"
```
Comment (davidvossel) [inline templates/cluster-template.yaml:71]: note that we had to make the criSocket templated to work with cri-o

File: config/default/manager_image_patch.yaml (hunk 1 above)
Comment (davidvossel) [reply config/default/manager_image_patch.yaml:11]: @rmohr do you think the right path forward here is to create a unique `/config/kubevirtci/` ?

Comment (rmohr) [reply config/default/manager_image_patch.yaml:11]: Yes sounds great : +1. We can the drop the 

```
make set-manifest-image
```

in the kubevirtci script.

Comment (davidvossel) [reply config/default/manager_image_patch.yaml]: fixed

**Comment by rmohr** [comment]
Successful CI run against this PR: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubernetes-sigs_cluster-api-provider-kubevirt/24/pull-kubernetes-sigs-cluster-api-provider-kubevirt-e2e/1466458307811086336

**Review by agradouski — APPROVED** [review]

**Pull request merged**